### PR TITLE
Fix Pomeg glitch for event eggs

### PR DESCRIPTION
**Pull request by dot-Comfey**

### CODE REVIEW

File: sim/team-validator.ts [hunk 1]
```diff
@@ -2476,6 +2498,10 @@ export class TeamValidator {
 							moveSources.add('1ST' + learned.slice(2) + ' ' + species.id);
 						}
 						moveSources.add(learned + ' ' + species.id);
+						const eventLearnset = dex.species.getLearnsetData(species.id);
+						if (eventLearnset.eventData && eventLearnset.eventData[parseInt(learned.charAt(2))].eventEgg && learnedGen === 3) {
```
Comment (KrisXV) [inline sim/team-validator.ts:2502]:
```suggestion
						if (eventLearnset.eventData?.[parseInt(learned.charAt(2))].eventEgg && learnedGen === 3) {
```

File: sim/global-types.ts [hunk 2]
```diff
@@ -85,6 +85,8 @@ interface EventInfo {
 	from?: string;
 	/** Japan-only events can't be transferred to international games in Gen 1 */
 	japan?: boolean;
+	/** For Gen 3 event eggs to allow Pomeg glitched moves */
```
Comment (DaWoblefet) [inline sim/global-types.ts:88]: I'd clarify it's specifically for Emerald (at least that's the assumption we have right now).

Comment (dot-Comfey) [reply sim/global-types.ts]: Unless another obscure mechanic regarding event egg moves is found, I think it's unlikely that it will see more general use.

**Comment by DaWoblefet** [comment]
Thanks dot-Comfey!

**Pull request merged**